src/midi/midi_fluidsynth.cpp

```diff
@@ -207,7 +207,17 @@ static std_fs::path find_sf_file(const std::string& name)
 			LOG_MSG("FSYNTH: FluidSynth checking if '%s' exists", sf.c_str());
 	#endif
 			if (path_exists(sf)) {
-				return sf;
+				// Parts of the path come from the `soundfont` setting, and
+				// `soundfont = FluidR3_GM.sf2` and `soundfont =
+				// fluidr3_gm.sf2` refer to the same file on case-preserving
+				// filesystems on Windows and macOS.
+				//
+				// `std_fs::canonical` returns the absolute path and matches
+				// its casing to that of the actual physical file. This
+				// prevents certain subtle bugs downstream when we use this
+				// path in comparisons.
+				//
+				return std_fs::canonical(sf).c_str();
```

Comment:
I'm not sure that `std_fs::cannoical` does anything with regards to casing.  At least https://en.cppreference.com/w/cpp/filesystem/canonical has no mention of it.

Reply:
Yeah, the standard does not mention it, but it's de-facto behaviour. I learned about it because many sources recommended it online.

It's similar to resizing a vector to a smaller size—the standard doesn't really say there will be no realloc involved, but virtually all sane standard libraries won't realloc when shrinking the vector, just update its size internal var.

As we're not aiming to run Staging on all sorts of esoteric C++ compilers and platforms that might implement `canonical` differently, I think it's ok to rely on such de-facto Clang/gcc behaviour.

Reply:
Yeah, cosmetic, as shown on the "current main behaviour" screenshot above. I'm pretty sure it works on Windows too, but I'll confirm.

Reply:
Confirmed, works on Windows too.